### PR TITLE
WIP Support custom data on all forms (preProcess)

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Testing ways to support customdata on all forms

@eileenmcnaughton Let's see if tests pass. What do you think?

We would need to implement similar for buildQuickForm and postProcess too.

### CODE REVIEW

File: CRM/Core/Form.php [hunk 1]
```diff
@@ -420,6 +420,21 @@ public function &add(
    * This is a virtual function and should be redefined if needed.
    */
   public function preProcess() {
+    try {
+      $entity = $this->getDefaultEntity();
+      if (empty($this->_id)) {
+        return;
```
Comment (eileenmcnaughton) [inline CRM/Core/Form.php]: this line shouldn't be in preProcess

Comment (mattwire) [reply CRM/Core/Form.php]: Haha! It was written as a quick proof of concept.  Intention was to check if $this->_id existed and not run the custom data stuff if it doesn't

Comment (eileenmcnaughton) [reply CRM/Core/Form.php]: I'm just trying to think of any risks from security POV of this - haven't yet but need to read the code more. Also, I wonder if we should also check the entity supports custom data.

Comment (mattwire) [reply CRM/Core/Form.php]: Ok, is there an easy way to check if the entity supports custom data?

Comment (eileenmcnaughton) [reply CRM/Core/Form.php]: CRM_Core_SelectValues::customGroupExtends()

Comment (eileenmcnaughton) [reply CRM/Core/Form.php]: (might be worth adding an isSupported helper - would save a couple of lines in more than one place probably & be good for readability)

**Comment by mattwire** [comment]
Superseded

**Pull request closed without merging**